vignettes/sensors.Rmd

```diff
@@ -16,140 +16,179 @@ knitr::opts_chunk$set(
+Let's plot to see how the accelerometer data actually looks like 
 
-```{r, echo=FALSE, fig.width=8, fig.height=4}
+```{r accel plots, echo=FALSE, fig.width=8, fig.height=4}
 library(ggplot2)
 a <- accelerometer_data
 a$t <- a$t - a$t[1]
 a <- tidyr::gather(a, 'axis' , 'val', -t)
 ggplot(a, aes(x = t, y= val)) + geom_line()+facet_wrap(~axis) + ylim(c(-0.025, 0.025))
 ```
+We are looking at 10s of data sampled at a high enough sampling rate(for accelerometer data) of 100Hz. The data looks noisy, and looks like it could make good use of some processing like frequency filters, maybe subsetting the time (time filtering) - all of which mhealthtools conveniently offers.
 
 #### Extracting features (using pre-existing pipeline)
-To use the existing feature set (you can define your own features or feed in your own model to extract features in `mhealthtools`!!). Note: You need to get the data in the proper format (as that of the sample data) to use the feature extraction pipeline.
+NOTE: To use the existing feature set (you can define your own features or feed in your own model to extract features in `mhealthtools`!!) you need to get the data in the proper format (as that of the sample data) to use the feature extraction pipeline.
 
-```{r, warning=FALSE, error=FALSE, prompt=FALSE}
+Let's take a look at the default feature extraction pipeline, whose features are all classical signal processing features based on time domain, frequency domain and the energy bands of the frequency spectrum.
+
+```{r accel default features, warning=FALSE, error=FALSE, prompt=FALSE}
 accel_features <- accelerometer_features(accelerometer_data)
 # using default parameters
 ```
+The output file is a list containing features extracted using supplied/default functions (extracted_features), features calculated using models(model_features) and an element for error (error). Let's look at the extracted_features to see how the default features look like.
 
-```{r}
-dplyr::glimpse(accel_features)
+```{r show accel default features}
+dplyr::glimpse(accel_features$extracted_features)
 ```
 
-`accelerometer_features` also offers multiple parameter options. The follwing examples are an effort to make the user aware of the options. 
+`accelerometer_features` also offers multiple parameter options. The following examples are an effort to make the user aware of the options. 
 
-In the following example, we change the window length (to 128, with 256 being the default), having the same overlap for the same `accel_data` used before
+In the following example, we change the window length (to 128, with NULL being the default - i.e no windowing), having the 50% overlap between consecutive windows for the same `accelerometer_data` used before
 
 ```{r, warning=FALSE, error=FALSE, prompt=FALSE}
 accel_features <- accelerometer_features(
   accelerometer_data,
   window_length = 128,
   window_overlap = 0.5)
-# Consider a window length of 128 samples
+# Consider a window length of 128 samples,
```

Comment:
I explicitly did not do that. 
I always found it effective to mention the parameters as Consider ... of ...

To write it as Extract features.. it would have to be a whole sentence like
Extract features over a window length of 128 samples with a window overlap of 0.5

Because two different sentences like
Extract features over a window length of 128
Extract features over a window overlap of 0.5

read like two different instances of feature extractions.

Reply:
> I always found it effective to mention the parameters as Consider ... of ...

Effective how? In my opinion, the phrasing is unconventional and not explicit. It is also redundant (it is clear from the function call that we are setting `window_length` to be 128, no need to comment afterwards `# Set/Consider a window length of 128 samples`).

Reply:
> > I always found it effective to mention the parameters as Consider ... of ...
> 
> Effective how? In my opinion, the phrasing is unconventional and not explicit. It is also redundant (it is clear from the function call that we are setting `window_length` to be 128, no need to comment afterwards `# Set/Consider a window length of 128 samples`).

The many texts I read reference parameters this way. 
I agree that it looks redundant, but I feel it is a good thing - only shows how good our parameters are named.